kolibri/deployment/default/cache.py

```diff
@@ -16,19 +16,21 @@
 diskcache_location = os.path.join(KOLIBRI_HOME, "process_cache")
 
 
-def recreate_cache():
-    shutil.rmtree(diskcache_location, ignore_errors=True)
-    os.mkdir(diskcache_location)
-    diskcache_cache = Cache(diskcache_location, disk_pickle_protocol=pickle_protocol)
-    diskcache_cache.close()
-    return diskcache_cache
-
-
-try:
-    diskcache_cache = Cache(diskcache_location, disk_pickle_protocol=pickle_protocol)
-    diskcache_cache.close()
-except DatabaseError:
-    diskcache_cache = recreate_cache()
+def recreate_diskcache():
```

Comment:
Do we really need this? are we using `.kolibri/process_cache/cache.db` anywhere?

Reply:
Yes - it is being used as the `"process_cache"` in Kolibri, in the case when we are not using Redis. Without this, then the RLock would not work cross process without Redis.

Reply:
I see the use of the database  in `CrossProcessCache` being used as a cache. 

However, for RLock, we're using `db_task_write_lock` and this one does not need the database after the changes this PR does.

Do we still need `CrossProcessCache` ? can't we use the `default` cache for the same task?

Reply:
If Redis is active, we are just using the default cache, but without Redis, the default cache is the InMemCache, which is not cross process. We want our database write lock to be active across processes, so this line: https://github.com/learningequality/kolibri/pull/6870/files#diff-f4bf89eeeed59f411ac09bca78e9e5dcR6 returns the cross process cache if it exists (i.e. when Redis is not available) and otherwise falls back to the default cache.

Reply:
Yes,  that makes totally sense. We should have either `redis` or `DiskCache` in CrossProcessCache for cache that need to be shared between processes.

But my doubts come for the fact that I have double checked that caches["process_cache"] is using InMemCache when running, for example, `kolibri manage importchannel network c401c87e3cf652128fd3efaa5b3774f2`

That's probably the reason why so many tests are failing. Something is not working properly and this code is not using DiskCache for process_cache

Reply:
Ah, ok, I must have messed something up somewhere. Will fix locally!

Reply:
Note that the issue above was due to the fact we overwrite all caches in the dev settings to LocMemCache - I have updated both the test and dev settings to preserve the process_cache when it is present, so that this is less confusing for testing and development.